stetho/src/main/java/com/facebook/stetho/inspector/network/AbstractAsyncPrettyPrinterFactory.java

```diff
@@ -0,0 +1,85 @@
+package com.facebook.stetho.inspector.network;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import com.facebook.stetho.common.ExceptionUtil;
+import com.facebook.stetho.common.Util;
+import com.facebook.stetho.inspector.network.AsyncPrettyPrinter;
+import com.facebook.stetho.inspector.network.AsyncPrettyPrinterFactory;
+
+import javax.annotation.Nullable;
+
+/**
+ * Abstract class for pretty printer factory that asynchronously downloads schema needed for
+ * pretty printing the payload
+ */
+public abstract class AbstractAsyncPrettyPrinterFactory implements AsyncPrettyPrinterFactory {
```

Comment:
This really should be renamed to something more deliberate like: `DownloadingAsyncPrettyPrinterFactory`.